tiled_layers/export-tiles/src/main/java/com/esri/samples/export_tiles/ExportTilesSample.java

```diff
@@ -105,97 +110,102 @@ public void start(Stage stage) {
         }
       });
 
-      // create button to export tiles
-      Button exportTilesButton = new Button("Export Tiles");
-
-      // create progress bar to show task progress
-      ProgressBar progressBar = new ProgressBar();
-      progressBar.setProgress(0.0);
-      progressBar.setVisible(false);
-
-      // when the button is clicked, export the tiles to a temporary file
-      exportTilesButton.setOnAction(e -> {
-        try {
-          // disable the button and show the progress bar
-          exportTilesButton.setDisable(true);
-          progressBar.setVisible(true);
-
-          // create a file and define the scale for the job
-          File tempFile = File.createTempFile("tiles", ".tpk");
-          double minScale = mapView.getMapScale();
-          double maxScale = tiledLayer.getMaxScale();
-
-          // minScale must always be larger than maxScale
-          if (minScale <= maxScale) {
-            minScale = maxScale + 1;
-          }
+      // when the map has loaded, create a tiled layer from it and export tiles
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
 
-          // create a task
-          ExportTileCacheTask exportTileCacheTask = new ExportTileCacheTask(tiledLayer.getUri());
+          // create a tiled layer from the basemap
+          ArcGISTiledLayer tiledLayer = (ArcGISTiledLayer) map.getBasemap().getBaseLayers().get(0);
 
-          // create parameters for the export tiles job
-          ListenableFuture<ExportTileCacheParameters> exportTileCacheParametersListenableFuture =
-                  exportTileCacheTask.createDefaultExportTileCacheParametersAsync(downloadArea.getGeometry(), minScale, maxScale);
+          // create progress bar to show task progress
+          var progressBar = new ProgressBar();
```

Comment:
You can instantiate a ProgressBar with the progress value like `new ProgressBar(0.0)`. But fair if you want to keep it separate!

Reply:
ooh! nice: I'll update